Lib/test/test_concurrent_futures.py

```diff
@@ -318,7 +318,7 @@ def test_threads_terminate(self):
         self.executor.submit(mul, 21, 2)
         self.executor.submit(mul, 6, 7)
         self.executor.submit(mul, 3, 14)
-        self.assertEqual(len(self.executor._threads), 3)
+        self.assertTrue(len(self.executor._threads) < 3)
```

Comment:
You've submitted three things, so why wouldn't it be possible for it to have launched three threads? Granted, the work per task is cheap, so it's possible (maybe even likely, given GIL interference) that the first task is done in time for its thread to be reused by the third task, but that's not guaranteed, is it?

Reply:
I agree that this test might fail randomly. It would be better to force the collection of the 3 tasks with calls to `result` before launching the next one and verify that only one thread has been used.